providers/image.rb

```diff
@@ -236,10 +253,9 @@ def remove
 
 def repository_and_tag_args
   docker_cmd_args = ''
-  if new_resource.repository
-    docker_cmd_args = new_resource.repository
-    docker_cmd_args += ":#{new_resource.tag}" if new_resource.tag
-  end
+  docker_cmd_args += new_resource.repository + '/' if new_resource.repository
+  docker_cmd_args += new_resource.image_name.split(':', 2).first # strip tag if any
```

Comment:
After checking, the `repository_and_tag_args` method could be simplified to leverage `image_and_tag_arg` and renamed to `repository_image_and_tag_arg` because the new name will be closer to what the method does. Want me to add to the PR?